src/openms_gui/include/OpenMS/VISUAL/DIALOGS/ToolsDialog.h

```diff
@@ -76,12 +75,13 @@ namespace OpenMS
         @brief Constructor
 
         @param parent Qt parent widget
+        @param params Containing all TOPP tool/util params
         @param ini_file The file name of the temporary INI file created by this dialog
         @param default_dir The default directory for loading and storing INI files
         @param layer_type The type of data (determines the applicable tools)
         @param layer_name The name of the selected layer
     */
-    ToolsDialog(QWidget * parent, String ini_file, String default_dir, LayerData::DataType layer_type, String layer_name);
+    ToolsDialog(QWidget * parent, Param params, String ini_file, String default_dir, LayerData::DataType layer_type, String layer_name);
```

Comment:
const &

Reply:
Currently the constructor is called with
```
ToolsDialog tools_dialog(this, param_.copy("tool_params:", true), topp_.file_name + "_ini", current_path_, layer.type, layer.getName());
```
So the copy does not live long enough to be passed as reference. Would you suggest just passing the whole `param_`?

Reply:
Temporaries should bind just fine to const references. They do not bind to non-const refs though. Did you use 'const Param& params'??

Reply:
You're right. I thought I had tried this before and it did not work.